File: flat-register-backend/src/main/java/com/flat/app/service/OwnerService.java
package com.flat.app.service;

import java.util.List;

import com.flat.app.entity.Owner;
import com.flat.app.exception.FlatNotFoundException;
import com.flat.app.exception.FlatRegisteredException;
import com.flat.app.exception.NoUsersFoundException;

public interface OwnerService {

	public List<Owner> getAllOwners() throws NoUsersFoundException;

	public Owner createOwner(Owner owner) throws FlatRegisteredException, FlatNotFoundException;

	public Owner getOwnerById(Long ownerId) throws NoUsersFoundException;

	public Owner getOwnerByFlatId(Long flatId) throws NoUsersFoundException;

	public Owner updateOwner(Long ownerId, Owner owner)
			throws FlatRegisteredException, FlatNotFoundException, NoUsersFoundException;

	public Owner getOwnerByOwnerEmail(String ownerEmail);

	public List<Owner> getOwnerByOwnerName(String ownerName);

}

File: flat-register-frontend/src/app/service/owner/owner.service.ts
import { Observable } from 'rxjs';
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

//const DOMAIN = 'http://localhost:8888'
const DOMAIN = 'https://flatregister.herokuapp.com'

@Injectable({
  providedIn: 'root'
})
export class OwnerService {

  private _baseURL = `${DOMAIN}/api/owner`;
  private _getOwnerByIdAPI = `${this._baseURL}/get-owner`;
  private _createOwnerAPI = `${this._baseURL}/register-flat`;
  private _getAllOwnersAPI = `${this._baseURL}/get-all-owners`;
  private _getOwnerByFlatIdAPI = `${this._baseURL}/get-flat-owner`;

  constructor(
    private _httpClient: HttpClient
  ) {
  }

  getAllOwners(): Observable<any> {
    return this._httpClient.get<any>(this._getAllOwnersAPI);
  }

  createOwner(request: any): Observable<any> {
    return this._httpClient.post<any>(this._createOwnerAPI, request);
  }

  getOwnerByFlatId(flatId: number): Observable<any> {
    return this._httpClient.get<any>(`${this._getOwnerByFlatIdAPI}/${flatId}`);
  }

  getOwnerById(ownerId: number): Observable<any> {
    return this._httpClient.get<any>(`${this._getOwnerByIdAPI}/${ownerId}`);
  }

  updateOwner(ownerId: any, request: any): Observable<any> {
    return this._httpClient.put<any>(`${this._createOwnerAPI}/${ownerId}`, request);
  }

}
